pkg/algorithm/cell_allocation.go

```diff
@@ -78,6 +79,70 @@ func buddyAlloc(
 	return false
 }
 
+// after buddyAlloc failed to allocate cells,
+// try to split a higher level cell to get current level cells.
+func splitAlloc(
```

Comment:
maybe we need a better and clearer name for this function...

Reply:
I can't come up with a good name, any suggestion?

Reply:
@hzhua @yqwang-ms any suggestion? 😂

Reply:
How about "badCellAvoidanceAlloc"?

Reply:
From the perspective of the rule of the algorithm, it indeed is a relaxed version of BuddyAlloc, as it allows to split a free cell higher than the lowest level. Maybe safeRelaxedBuddyAlloc?

Reply:
@zhypku, sounds good to me.

Reply:
updated to `safeRelaxedBuddyAlloc`

Reply:
Is it possible that a safe split will trigger the merge operation of buddyAlloc? e.g., suppose a L2 cell's L1 cells are all bad cells. the safeRelaxedBuddyAlloc will split the L2 cell (named A), and go ahead to split another L2 cell (B). Since all the L1 cells of A are free cells (albeit bad), the BuddyAlloc will merge them into a L2 cell (i.e., A).
My question is, when will buddyAlloc do the merging?

Reply:
The free list for cell allocation is not persistent. It is just a copy of the ground truth free list. So even safeRelaxedBuddyAlloc splits a cell, it is not effective immediately. It is effective when we confirm this allocation. In your example, what we truly allocate is an L1 cell from B. So we only split B in the ground truth free list, and won't split A.